ARROW-16554: [Java] Download Java nightlies artifacts from https://nightlies.apache.org/arrow/java/org/apache/arrow/ (#13352)
apache/arrow: docs/source/java/install.rst

Arrow nightly builds are posted on the mailing list at `builds@arrow.apache.org`_.
The artifacts are uploaded to GitHub. For example, for 2022/03/01, they can be found at `Github Nightly`_.

Installing from Apache Nightlies
********************************

Steps involved to install to local Maven repository using Apache Nightlies Repository:

1. Is needed to add Apache Nightlies Repository to the Maven/Gradle project. For Maven pom.xml is needed to add arrow/java directory.

.. code-block:: xml

    <properties>

>         <arrow.version>9.0.0.dev191</arrow.version>

Reviewer: Does the nightlies directory contain only 1 nightly build? Shouldn't it contain multiple nightly builds? I think it was mentioned to keep around a few weeks of them.

In such case, how does the user know which version to put here? We might want to write somewhere that you are expected to go to https://nightlies.apache.org/arrow/java/org/apache/arrow/arrow-memory/ (for example) and see which versions are vailable, or have some other way to know which is the right version.

Author: This is an incremental directory, it means we haver more Java artifacts as more nightly build do we have. Pending to define what could the maximum versions storage.

Adding more detail on the steps.

Arrow nightly builds are posted on the mailing list at `builds@arrow.apache.org`_.
The artifacts are uploaded to GitHub. For example, for 2022/03/01, they can be found at `Github Nightly`_.

Installing from Apache Nightlies
********************************

> 1. Look up the nightly version number for the Arrow libraries used.
> 
>     For example, for ``arrow-memory``, visit  https://nightlies.apache.org/arrow/java/org/apache/arrow/arrow-memory/ and see what versions are available (e.g. 9.0.0.dev191).

Reviewer 2: ```suggestion
1. Look up the nightly version number for the Arrow libraries used.

   For example, for ``arrow-memory``, visit  https://nightlies.apache.org/arrow/java/org/apache/arrow/arrow-memory/ and see what versions are available (e.g. 9.0.0.dev191).
```

Arrow nightly builds are posted on the mailing list at `builds@arrow.apache.org`_.
The artifacts are uploaded to GitHub. For example, for 2022/03/01, they can be found at `Github Nightly`_.

Installing from Apache Nightlies
********************************

> 
> Steps involved to download Java artifacts from Apache Nightlies Repository:
> 

Reviewer 2: ```suggestion

```

> 1. Once you have the Java nightly library needed (i.e. arrow-memory), please go to https://nightlies.apache.org/arrow/java/org/apache/arrow/arrow-memory/ and see what version are available (i.e. 9.0.0.dev191), this is alligned with Arrow nightly builds.

Reviewer 2: ```suggestion
1. Look up the nightly version number for the Arrow libraries used.

    For example, for ``arrow-memory``, visit  https://nightlies.apache.org/arrow/java/org/apache/arrow/arrow-memory/ and see what versions are available (e.g. 9.0.0.dev191).
```

Reviewer 2: (not sure about the reST formatting here, please correct)

2. Add Apache Nightlies Repository to the Maven/Gradle project.

.. code-block:: xml

    <properties>
        <arrow.version>9.0.0.dev191</arrow.version>
    </properties>
    ...
    <repositories>
        <repository>
            <id>arrow-apache-nightlies</id>
            <url>https://nightlies.apache.org/arrow/java</url>
        </repository>
    </repositories>
    ...
    <dependencies>
        <dependency>
            <groupId>org.apache.arrow</groupId>
            <artifactId>arrow-vector</artifactId>
            <version>${arrow.version}</version>
        </dependency>
    </dependencies>
    ...

Installing Manually
*******************

> 
> Steps involved to install to local Maven repository manually:

Reviewer 2: ```suggestion
```
